apps/ae_socket_connector/lib/client_runner.ex

```diff
@@ -0,0 +1,236 @@
+defmodule ClientRunner do
+  use GenServer
+  require Logger
+
+  defstruct pid_session_holder: nil,
+            color: nil,
+            job_list: nil
+
+  def start_link(
+        {_pub_key, _priv_key, %SocketConnector.WsConnection{}, _ae_url, _network_id, _role, _jobs,
+         _color} = params
```

Comment:
`params` should be a map, which helps make the API more readable. Also why are you checking only for the struct here?